foursight_core/react/api/react_api.py

```diff
@@ -0,0 +1,447 @@
+        VALID/authenticated JWT; if this were not so we would have returned before this call,
+        in app_utils/auth_callback. We create a new JWT from the given JWT, which we call authtoken,
+        and set this as a cookie on the redirect (HTTP 302) response which we return.
+        The jwt_expires_in is the number of seconds from now when the given JWT will expire;
+        this is directly from the (above mentioned) Auth0 POST; this is NOT the exp from
+        the JWT which seems to be different from (like 14 hours less than) jwt_expires_in.
+        """
+        jwt_expires_at = convert_datetime_to_time_t(datetime.datetime.utcnow() + datetime.timedelta(seconds=jwt_expires_in))
+        authtoken = self.auth.create_authtoken(jwt, jwt_expires_at, env, domain)
+        authtoken_cookie = create_set_cookie_string(request, name="authtoken",
+                                                    value=authtoken,
+                                                    domain=domain,
+                                                    expires=jwt_expires_at, http_only=False)
+        redirect_url = self._get_redirect_url(request, env, domain, context)
+        return self.create_redirect_response(redirect_url, {"set-cookie": authtoken_cookie})
+
+    def _get_redirect_url(self, request: dict, env: str, domain: str, context: str):
+        redirect_url = read_cookie(request, "reactredir")
+        if not redirect_url:
+            is_running_locally = app.core.is_running_locally(request)
+            if is_running_locally:
+                http = "http"
+                # Using ROUTE_PREFIX here instead of context because may be just "/" for the local
+                # deploy case because we get here via the /callback endpoint (during authentication).
+                # Due to confusion between local deploy not implicitly using /api as context so setting
+                # it explicitly; this is just so we are dealing with the same paths for either case.
+                context = ROUTE_PREFIX
+            else:
+                http = "https"
+            if not env:
+                env = self.envs.get_default_env()
+            if not context:
+                context = "/"
+            elif not context.endswith("/"):
+                context = context + "/"
+            redirect_url = f"{http}://{domain}{context}react/{env}/login"
+        else:
+            # Not certain if by design but the React library (universal-cookie) used to
+            # write cookies URL-encodes them; rolling with it for now and URL-decoding here.
+            redirect_url = urllib.parse.unquote(redirect_url)
+        return redirect_url
+
+    def react_authorize(self, request: dict, env: str) -> dict:
+        return self.auth.authorize(request, env)
+
+    def react_serve_static_file(self, env: str, **kwargs):
+        return self.react_ui.serve_static_file(env, **kwargs)
+
+    def reactapi_logout(self, request: dict, env: str):
```

Comment:
Generally I'm not sure it's necessary to prefix all these methods with `react` or `reactapi` since you're already in a class called `ReactAPI`?